.github/workflows/java-build.yml

```diff
@@ -1,24 +1,61 @@
 name: Build Java Semantic Kernel
 
 on:
+  workflow_dispatch:
   push:
     branches: [ "experimental-java", "java-dev" ]
+    paths:
+      - 'java/**'
   pull_request:
     branches: [ "experimental-java", "java-dev" ]
+    paths:
+      - 'java/**'
 
-jobs:
-  build:
+permissions:
+  contents: read
 
+jobs:
+  java-build:
     runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        java-versions: [8, 11, 17]
```

Comment:
Hey @johnoliver, should @Luigi96 here update the build pipeline so that the library is built using JDK 17, with target for Java 8?

One thing I believe we would have to do though, is run tests with JDK 8, unless there is a guaranteed way to prevent us from using Java 17 APIs (types/methods).

If it is ok to have 8 for building, should we simplify and remove 11? Or do we need all three?

Reply:
I would prefer to build with 8 and at least one lts>9, (i.e 11, 17). I don't _think_ there is going to be much difference discovered between 11 and 17 build so I am fine with removing 11

Reply:
Alright, so after our meeting today, let's move forward with JDK 8 and JDK 17 only, in the build pipelines.